File: lib/repositories/dog_repository.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../models/dog_image.dart';

class DogRepository {
  final String baseUrl = 'https://dog.ceo/api/breeds/image/random';

  Future<DogImage> getRandomDogImage() async {
    try {
      final response = await http.get(Uri.parse(baseUrl));
      
      if (response.statusCode == 200) {
        return DogImage.fromJson(json.decode(response.body));
      } else {
        throw Exception('Failed to load dog image');
      }
    } catch (e) {
      throw Exception('Failed to connect to the API');
    }
  }
} 

File: app/src/main/java/com/example/dogapp/repository/DogRepository.kt
package com.example.dogapp.repository

import com.example.dogapp.model.DogImage
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET

interface DogApi {
    @GET("breeds/image/random")
    suspend fun getRandomDogImage(): DogImage
}

class DogRepository {
    private val api: DogApi = Retrofit.Builder()
        .baseUrl("https://dog.ceo/api/")
        .addConverterFactory(GsonConverterFactory.create())
        .build()
        .create(DogApi::class.java)

    suspend fun getRandomDogImage(): DogImage {
        return try {
            api.getRandomDogImage()
        } catch (e: Exception) {
            throw Exception("Failed to load dog image")
        }
    }
} 